third_party/java/arcs/build_defs/internal/tools.oss.bzl

```diff
@@ -15,21 +19,73 @@ def arcs_manifest_parse_test(name, srcs, deps = []):
         if not src.endswith(".arcs"):
             fail("src must be an .arcs manifest file, found %s" % src)
 
-    manifest_name = name + "_sources"
+    test_args = " ".join(["--src $(location %s)" % src for src in srcs])
 
-    arcs_manifest(
-        name = manifest_name,
+    sigh_command(
+        name = name,
         srcs = srcs,
         deps = deps,
+        sigh_cmd = "run manifestChecker " + test_args,
+        progress_message = "Checking Arcs manifest",
+        execute = False,
     )
 
-    test_args = " ".join(["--src $(location %s)" % src for src in srcs])
+# buildifier: disable=function-docstring
+def arcs_tool_recipe2plan(name, srcs, outs, deps, generate_proto = False, recipe = None):
+    sigh_cmd = "recipe2plan --quiet --outdir $(dirname {OUT}) --outfile $(basename {OUT})"
+    if generate_proto:
+        sigh_cmd += " --format proto"
+    if recipe:
+        sigh_cmd += " --recipe " + recipe
+    sigh_cmd += " {SRC}"
 
     sigh_command(
         name = name,
         srcs = srcs,
-        deps = deps + [":" + manifest_name],
-        sigh_cmd = "run manifestChecker " + test_args,
-        progress_message = "Checking Arcs manifest",
-        execute = False,
+        outs = outs,
+        progress_message = "Generating Kotlin Plans",
```

Comment:
Optional: Modify the msg to indicate if we're generating Kotlin Plans or protos.

Reply:
Done